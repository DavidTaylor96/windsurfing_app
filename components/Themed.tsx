import * as React from "react";
import { Text as DefaultText, View as DefaultView } from "react-native";

import Colors from "../constants/Colors";
import useColorScheme from "../hooks/useColorScheme";

export function useThemeColor(
  props: { light?: string; dark?: string },
  colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) {
  const theme = useColorScheme();
  const colorFromProps = props[theme];

  if (colorFromProps) {
    return colorFromProps;
  } else {
    return Colors[theme][colorName];
  }
}

type ThemeProps = {
  lightColor?: string;
  darkColor?: string;
};

type FontWeightProps = {
  weight?: "normal" | "h1" | "h2" | "h3" | "h4" | "h5";
};

export type TextProps = FontWeightProps & ThemeProps & DefaultText["props"];
export type ViewProps = ThemeProps & DefaultView["props"];

export function Text(props: TextProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const color = useThemeColor({ light: lightColor, dark: darkColor }, "text");

  let fontSize = 14;
  let fontFamily:
    | "sora-bold"
    | "sora-extra-bold"
    | "sora-extra-light"
    | "sora-light"
    | "sora-medium"
    | "sora-regular"
    | "sora-semi-bold"
    | "sora-thin" = "sora-regular";

  switch (props.weight) {
    case "normal":
      fontSize = 12;
      break;
    case "h1":
      fontSize = 24;
      break;
    case "h2":
      fontSize = 24;
      fontFamily = "sora-semi-bold";
      break;
    case "h3":
      fontSize = 18;
      break;
    case "h4":
      fontSize = 16;
      break;
    case "h5":
      fontSize = 16;
      fontFamily = "sora-semi-bold";

      break;
    default:
      fontSize = 12;
  }

  return (
    <DefaultText
      style={[{ fontSize, fontFamily }, { color }, style]}
      {...otherProps}
    />
  );
}

export function View(props: ViewProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "background"
  );

  return <DefaultView style={style} {...otherProps} />;
}
