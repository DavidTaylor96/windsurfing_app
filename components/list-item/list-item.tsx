import * as React from 'react';
import { FC } from 'react';
import { StyleSheet, Image, Pressable } from 'react-native';

interface IListItem {
  onPress: () => void;
}

export const ListItem: FC<IListItem> = (props) => {
  return (
    <Pressable onPress={props.onPress} style={styles.wrapper}>
      {props.children}
    </Pressable>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    paddingVertical: 20,
    borderBottomWidth: 1,
    borderColor: '#CBCBCB'
  },
});
