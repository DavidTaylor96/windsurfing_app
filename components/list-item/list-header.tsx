import * as React from 'react';
import { FC } from 'react';
import { StyleSheet } from 'react-native';
import { Text, View } from '../Themed';

interface IListHeader {}

export const ListHeader: FC<IListHeader> = (props) => {
  return (
    <View style={styles.header}>
      <Text weight="h1">{props.children}</Text>
      <View style={styles.underline} />
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    backgroundColor: 'white',
    marginHorizontal: 20,
    paddingVertical: 10,
  },

  underline: {
    height: 2,
    width: 40,
    backgroundColor: '#F401F2',
    marginTop: 3,
  },
});
