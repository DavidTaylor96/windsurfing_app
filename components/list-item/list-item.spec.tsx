import * as React from 'react';
import renderer from 'react-test-renderer';
import { ListItem } from './list-item';
import { Text } from '../Themed'


it(`List Item`, () => {

  const pressEvent = jest.fn();

  const listItem = renderer.create(
      <ListItem onPress={pressEvent}>
        <Text>Text</Text>
      </ListItem>

  );


  listItem.root.props.onPress();
  
  expect(pressEvent).toHaveBeenCalledTimes(1);
  expect(listItem.root.children.length).toBe(1)
  expect(listItem).toMatchSnapshot();
});
