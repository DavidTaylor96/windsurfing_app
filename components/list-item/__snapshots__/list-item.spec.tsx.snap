// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "borderColor": "#E5E5E5",
      "flexDirection": "row",
      "marginHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "sora-regular",
          "fontSize": 12,
        },
        Object {
          "color": "black",
        },
        undefined,
      ]
    }
  >
    Text
  </Text>
</View>
`;
